rest-service/manager_rest/rest/resources_v3/events.py

```diff
@@ -50,22 +51,37 @@ def post(self):
             check_user_action_allowed('set_timestamp')
 
         sm = get_storage_manager()
-        exc = current_execution._get_current_object()
-        if exc is None:
-            exc_id = request_dict.get('execution_id')
-            if exc_id is None:
+
+        exc_id = request_dict.get('execution_id')
+        eg_id = request_dict.get('execution_group_id')
+
+        if exc_id and eg_id:
+            raise manager_exceptions.ConflictError(
+                'Both execution_id and execution_group_id were provided.')
+
+        exc_params = {}
+        if eg_id:
+            eg = sm.get(models.ExecutionGroup, eg_id)
+            exc_params['_execution_group_fk'] = eg._storage_id
+        else:
+            if exc_id:
+                exc = sm.get(models.Execution, exc_id)
+            else:
+                exc = current_execution._get_current_object()
+
+            if exc is None:
                 raise manager_exceptions.ConflictError(
-                    'No execution passed, and not authenticated by '
-                    'an execution token')
-            exc = sm.get(models.Execution, request_dict.get('execution_id'))
-        exc_params = {
-            '_execution_fk': exc._storage_id,
+                     'No execution passed, and not authenticated by '
+                     'an execution token')
+
+            exc_params['_execution_fk'] = exc._storage_id
+        exc_params.update({
             '_tenant_id': exc._tenant_id,
```

Comment:
if `eg_id` is set, then `exc` is just a NameError?

Reply:
if eg_id is set you can't reach this path.

Reply:
seems to me you can!

Reply:
Good catch, updated.